File: app/src/main/java/com/mikelau/cropme/MainActivity.java
package com.mikelau.cropme;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.mikelau.croperino.Croperino;
import com.mikelau.croperino.CroperinoConfig;
import com.mikelau.croperino.CroperinoFileUtil;

public class MainActivity extends AppCompatActivity {

    private Button btnSummon;
    private ImageView ivMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnSummon = (Button) findViewById(R.id.btn_click);
        ivMain = (ImageView) findViewById(R.id.iv_main);

        new CroperinoConfig("IMG_" + System.currentTimeMillis() + ".jpg", "/MikeLau/Pictures", "/sdcard/MikeLau/Pictures");
        CroperinoFileUtil.verifyStoragePermissions(MainActivity.this);
        CroperinoFileUtil.setupDirectory(MainActivity.this);

        btnSummon.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Croperino.prepareChooser(MainActivity.this, "Capture photo...", ContextCompat.getColor(MainActivity.this, android.R.color.background_dark));
            }
        });
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        switch (requestCode) {
            case CroperinoConfig.REQUEST_TAKE_PHOTO:
                if (resultCode == Activity.RESULT_OK) {
                    Croperino.runCropImage(CroperinoFileUtil.getmFileTemp(), MainActivity.this, true, 1, 1, 0, 0);
                }
                break;
            case CroperinoConfig.REQUEST_PICK_FILE:
                if (resultCode == Activity.RESULT_OK) {
                    CroperinoFileUtil.newGalleryFile(data, MainActivity.this);
                    Croperino.runCropImage(CroperinoFileUtil.getmFileTemp(), MainActivity.this, true, 1, 1, 0, 0);
                }
                break;
            case CroperinoConfig.REQUEST_CROP_PHOTO:
                if (resultCode == Activity.RESULT_OK) {
                    Uri i = Uri.fromFile(CroperinoFileUtil.getmFileTemp());
                    ivMain.setImageURI(i);
                    //Do saving / uploading of photo method here.
                }
                break;
            default:
                break;
        }
    }
}

File: app/src/main/java/com/mikelau/croperinosample/MainActivity.kt
package com.mikelau.croperinosample

import android.Manifest
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.lorenzofelletti.permissions.PermissionManager
import com.lorenzofelletti.permissions.dispatcher.dsl.checkPermissions
import com.lorenzofelletti.permissions.dispatcher.dsl.doOnDenied
import com.lorenzofelletti.permissions.dispatcher.dsl.doOnGranted
import com.lorenzofelletti.permissions.dispatcher.dsl.showRationaleDialog
import com.lorenzofelletti.permissions.dispatcher.dsl.withRequestCode
import com.mikelau.croperino.Croperino
import com.mikelau.croperino.CroperinoConfig
import com.mikelau.croperino.CroperinoFileUtil

class MainActivity : AppCompatActivity() {

    lateinit var btnCamera: Button
    lateinit var btnGallery: Button
    lateinit var ivMain: ImageView
    lateinit var permissionManager: PermissionManager

    companion object {
        private const val POSITION_REQUEST_CODE = 1
        private val POSITION_REQUIRED_PERMISSIONS = arrayOf(
            Manifest.permission.CAMERA,
            Manifest.permission.READ_EXTERNAL_STORAGE,
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.MANAGE_EXTERNAL_STORAGE,
            Manifest.permission.READ_MEDIA_IMAGES,
            Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnCamera = findViewById(R.id.btn_camera)
        btnGallery = findViewById(R.id.btn_gallery)
        ivMain = findViewById(R.id.iv_main)

        permissionManager = PermissionManager(this)

        permissionManager buildRequestResultsDispatcher {
            withRequestCode(POSITION_REQUEST_CODE) {
                checkPermissions(POSITION_REQUIRED_PERMISSIONS)
                showRationaleDialog(message = "Please accept all permission to use Croperino Library")
                doOnGranted {
                }
                doOnDenied {
                }
            }
        }

        permissionManager checkRequestAndDispatch POSITION_REQUEST_CODE
        permissionManager.checkRequestAndDispatch(
            POSITION_REQUEST_CODE
        )

        CroperinoConfig(
            "IMG_" + System.currentTimeMillis() + ".jpg",
            "/MikeLau/Pictures/",
            "/sdcard/MikeLau/Pictures/"
        )
        CroperinoFileUtil.setupDirectory(this)

        btnCamera.setOnClickListener {
            Croperino.prepareCamera(this)
        }

        btnGallery.setOnClickListener {
            Croperino.prepareGallery(this)
        }
    }

    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            CroperinoConfig.REQUEST_TAKE_PHOTO -> if (resultCode == RESULT_OK) {
                Croperino.runCropImage(CroperinoFileUtil.tempFile!!, this, true, 1, 1,
                    com.mikelau.croperino.R.color.gray,
                    com.mikelau.croperino.R.color.gray_variant
                )
            }

            CroperinoConfig.REQUEST_PICK_FILE -> if (resultCode == RESULT_OK) {
                CroperinoFileUtil.newGalleryFile(data!!, this@MainActivity)
                Croperino.runCropImage(
                    CroperinoFileUtil.tempFile!!,
                    this@MainActivity,
                    true,
                    1,
                    1,
                    com.mikelau.croperino.R.color.gray,
                    com.mikelau.croperino.R.color.gray_variant
                )
            }

            CroperinoConfig.REQUEST_CROP_PHOTO -> if (resultCode == RESULT_OK) {
                val i = Uri.fromFile(CroperinoFileUtil.tempFile)
                ivMain.setImageURI(i)
            }

            else -> {}
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int, permissions: Array<out String>, grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        permissionManager.dispatchOnRequestPermissionsResult(requestCode, grantResults)
    }
}
